### ISSUE TITLE
UP044 false positive around PEP 692 TypedDict `**kwargs` type hints

### ISSUE THREAD

**Issue by kkom** [astral-sh/ruff#14047]
Since ruff 0.7.2 the code below:

```python
class KwargsDict(TypedDict):
    foo: int
    bar: int

def fun(
    **kwargs: Unpack[KwargsDict],
) -> None: ...
```

Triggers this error:

```
UP044 [*] Use `*` for unpacking
```

I don't think the star syntax is allowed in this case though, see: https://peps.python.org/pep-0692/#introducing-a-new-syntax

And making this change triggers another Ruff error:

```
SyntaxError: Starred expression cannot be used here Ruff
```

**Comment by MichaReiser** [comment]
Thanks for reporting this and linking to the relevant PEP section. @diceroll123 are you interested in submitting a fix for the new rule?
